pkg/occlient/monitor.go

```diff
@@ -0,0 +1,171 @@
+package occlient
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/openshift/odo/pkg/log/fidget"
+	"github.com/pkg/errors"
+
+	appsv1 "github.com/openshift/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Error channels for both the pod and deploymentconfig
+var deploymentConfigErrorChannel chan error
+var podErrorChannel chan error
+
+// Main wait group for both go routines.
+var wg sync.WaitGroup
+
+// waitForDD is a go routine to wait for the DeploymentConfig to successfully complete / come up
+func (c *Client) waitForDC(name string, deploymentConfigSpinner *fidget.Spinny) {
```

Comment:
Why you couldn't use already existing `occlient.WaitAndGetDC` function? It has almost the same logic.